src/java/main/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -146,14 +146,26 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
             );
             containerManager.start();
 
+            // Watch status of ZooKeeper server. If there is an internal error
+            // then will do a graceful shutdown.
+            while (zkServer.isRunning()) {
```

Comment:
Agreed, I understand it. Instead of an interface I will directly make a ZooKeeperServerStateListener class and register like zks#registerStateListener(ZooKeeperServerStateListener zksStateListener);

OK?